packages/common/core-interfaces/src/handles.ts

```diff
@@ -102,18 +102,18 @@ export interface IFluidHandleInternal<
 }
 
 /**
+ * TODO: Better name to signal that this might be a placeholder - it's just placeholder-aware
  * @internal
  */
-export interface IFluidHandleInternalWithMetadata<
+export interface IFluidHandleInternalPlaceholder<
 	// REVIEW: Constrain `T` to something? How do we support dds and datastores safely?
 	out T = unknown, // FluidObject & IFluidLoadable,
 > extends IFluidHandleInternal<T> {
 	/**
-	 * The handle may contain metadata, generated and interpreted by the subsystem that the handle
-	 * relates to.  For instance, the BlobManager uses this to distinguish blob handles which may
-	 * not yet have an attached blob yet.
+	 * Whether the handle is a placeholder, meaning that it may exist before its payload is retrievable.
+	 * For instance, the BlobManager can generate handles before completing the blob upload/attach.
 	 */
-	readonly metadata: Readonly<Record<string, number | boolean | string>> | undefined;
+	readonly placeholder: boolean;
```

Comment:
i'd prefer we keep this wrapped in some object, as it allows for expandability without polluting the main object with miscellaneous properties

Reply:
Since this is core to the concept of the handle (i.e. can the payload be expected to exist) and not specific to a subsystem (e.g. is this a blob) it makes more sense directly on the object.

If we were to offer subsystem extensibility I think we'd want to add an `unknown` member for the subsystem to unpack, but here I intend to have the ContainerRuntime itself rather than a subsystem drive the placeholder logic (as in the linked branch)

Reply:
keeping this isolated will aid in extensibility in the future with almost no cost today, whereas not doing it will make extensibility and compact much harder to manage in the future. when it comes to data a rest we have long term support guarantees and wrapping this in an object makes those easier to meet regardless of what happens. Specifically, its aid in identification and transformation.

I'm also not convinced that this concept is broadly useful, it may be, but i don't see scenarios beyond blobs where it is immediately useful.

Reply:
We wouldn't make this an extensibility point - as I mention if we want subsystems to be able to extend the handle then we can later add that as an `unknown` extensibility point for the CR to pass on to them.  It would be inappropriate to mix `placeholder` with subsystem-specific `metadata` because my intent is for the placeholder handling to be managed in the CR as in the linked branch.

Wrapping it in an object also doesn't have any meaningful effect on long-term support guarantees, it's just an extra level of hierarchy but at the end of the day it's still a property on the handle.

Reply:
I'm not talking about external extensibility. I'm talking about internal extensibility.

Reply:
Me too

Reply:
what is the harm caused by nesting this under an object?